src/core_plugins/kibana/public/dashboard/get_top_nav_config.js

```diff
@@ -0,0 +1,84 @@
+import { DashboardViewMode } from './dashboard_view_mode';
+import { createTopNavExecuteConfig, createTopNavTemplateConfig } from 'ui/kbn_top_nav/kbn_top_nav_config';
+
+/**
+ * @param {DashboardMode} dashboardMode.
+ * @param kbnUrl - used to change the url.
+ * @param {(DashboardMode) => void} modeChange - a function to trigger a dashboard mode change.
+ * @return {Array} - Returns an array of objects for a top nav configuration, based on the
+ * mode.
+ */
+export function getTopNavConfig(dashboardMode, kbnUrl, modeChange) {
+  switch (dashboardMode) {
+    case DashboardViewMode.VIEW:
+      return [getNewConfig(kbnUrl), getOpenConfig(), getShareConfig(), getEditConfig(modeChange)];
+    case DashboardViewMode.EDIT:
+      return [getNewConfig(kbnUrl), getOpenConfig(), getAddConfig(), getSaveConfig(), getOptionsConfig(), getViewConfig(modeChange)];
+    default:
+      return [];
+  }
+}
+
+function getEditConfig(modeChange) {
+  return createTopNavExecuteConfig(
+    'edit',
+    'Switch to edit mode',
+    'dashboardEditMode',
+    () => { modeChange(DashboardViewMode.EDIT); });
+}
+
+function getViewConfig(modeChange) {
+  return createTopNavExecuteConfig(
+    'Stop Editing',
```

Comment:
Sorry, please disregard this entirely. I made this comment a while ago, and it's no longer valid.